src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Windows.cs

```diff
@@ -0,0 +1,175 @@
+            {
+                return false;
+            }
+
+            // Remember our neutrality
+            _bNeutral = *((uint*)pBuffer) != 0;
+
+            // Note: Parents will be set dynamically
+
+            // Start by assuming the windows name will be the same as the specific name since windows knows
+            // about specifics on all versions. Only for downlevel Neutral locales does this have to change.
+            _sWindowsName = realNameBuffer;
+
+            // Neutrals and non-neutrals are slightly different
+            if (_bNeutral)
+            {
+                // Neutral Locale
+
+                // IETF name looks like neutral name
+                _sName = realNameBuffer;
+
+                // Specific locale name is whatever ResolveLocaleName (win7+) returns.
+                // (Buffer has our name in it, and we can recycle that because windows resolves it before writing to the buffer)
+                result = Interop.Kernel32.ResolveLocaleName(realNameBuffer, pBuffer, Interop.Kernel32.LOCALE_NAME_MAX_LENGTH);
+
+                // 0 is failure, 1 is invariant (""), which we expect
+                if (result < 1)
+                {
+                    return false;
+                }
+                // We found a locale name, so use it.
+                // In vista this should look like a sort name (de-DE_phoneb) or a specific culture (en-US) and be in the "pretty" form
+                _sSpecificCulture = new string(pBuffer, 0, result - 1);
+            }
+            else
+            {
+                // Specific Locale
+
+                // Specific culture's the same as the locale name since we know its not neutral
+                // On mac we'll use this as well, even for neutrals. There's no obvious specific
+                // culture to use and this isn't exposed, but behaviorally this is correct on mac.
+                // Note that specifics include the sort name (de-DE_phoneb)
+                _sSpecificCulture = realNameBuffer;
+
+                _sName = realNameBuffer;
+
+                // We need the IETF name (sname)
+                // If we aren't an alt sort locale then this is the same as the windows name.
+                // If we are an alt sort locale then this is the same as the part before the _ in the windows name
+                // This is for like de-DE_phoneb and es-ES_tradnl that hsouldn't have the _ part
+
+                result = GetLocaleInfoEx(realNameBuffer, Interop.Kernel32.LOCALE_ILANGUAGE | Interop.Kernel32.LOCALE_RETURN_NUMBER, pBuffer, sizeof(int) / sizeof(char));
+                if (result == 0)
+                {
+                    return false;
+                }
+
+                _iLanguage = *((int*)pBuffer);
+
+                if (!IsCustomCultureId(_iLanguage))
+                {
+                    // not custom locale
+                    int index = realNameBuffer.IndexOf('_');
+                    if (index > 0 && index < realNameBuffer.Length)
```

Comment:
How could `index < realNameBuffer.Length` ever be false?

Reply:
I don't know, if we're getting the index from the same buffer, I don't expect `IndexOf` to return a index out of the string length 😄 -- I'm just copying over code from one file to another, but I can remove it as part of this change.